src/lib/common/clockFunctions.cpp

```diff
@@ -32,7 +32,7 @@
 *
 * clock_difftime - 
 */
-void clock_difftime(struct timespec* endTime, struct timespec* startTime, struct timespec* diffTime)
+void clock_difftime(const struct timespec* endTime, const struct timespec* startTime, struct timespec* diffTime)
```

Comment:
I understand that endTime and startTime are input parametesr and diffTime output parameter, so the best signature would be:

```
clock_difftime(const struct timespec& endTime, const struct timespec& startTime, struct timespec* diffTime)
```

Although I'm not sure if is possible, as startTime and endTime come as pointer from a lower-level function...